core/src/main/java/com/linecorp/armeria/server/AbstractVirtualHostBuilder.java

```diff
@@ -129,6 +129,7 @@
     private SslContext sslContext;
     @Nullable
     private Function<Service<HttpRequest, HttpResponse>, Service<HttpRequest, HttpResponse>> decorator;
+    private Function<VirtualHost, Logger> accessLoggerMapper = host -> null;
```

Comment:
I find this a little bit weird because we allow that virtual host level mapper to return null but we do not for the server-level mapper. How about making this `@Nullable` back again and set `null` to this field as the default value? `ServerBuilder.build()` could use the server-level mapper if the virtual host-level mapper is `null`, and otherwise the virtual host-level mapper could be used. WDYT?